### ISSUE TITLE
Script exits with failed exit code.

### ISSUE THREAD

**Issue by tayken** [tayken/ESP32-Auto-Programmer#2]
When the programming script is done, the service status changes to `failed` instead of some other successful state. This may cause issues with monitoring or other things down the line. Need to figure out why this is happening and take care of it.

```
× esp32-auto-program@sys-devices-pci0000:00-0000:00:14.0-usb1-1\x2d2-1\x2d2:1.0-tty-ttyACM0.service - Program ESP32 device attached to 'sys/devices/pci0000:00/0000:00:14.0/usb1/1-2/1-2:1.0/tty/ttyACM0'
     Loaded: loaded (/etc/systemd/system/esp32-auto-program@.service; static)
     Active: failed (Result: exit-code) since Wed 2024-11-06 15:03:50 JST; 6s ago
    Process: 115095 ExecStart=/home/tayken/esp32/build/esp32-program-shim.sh sys/devices/pci0000:00/0000:00:14.0/usb1/1-2/1-2:1.0/tty/ttyACM0 (code=exited, status=1/FAILURE)
   Main PID: 115095 (code=exited, status=1/FAILURE)
        CPU: 659ms

Nov 06 15:03:30 wadew systemd[1]: Started Program ESP32 device attached to 'sys/devices/pci0000:00/0000:00:14.0/usb1/1-2/1-2:1.0/tty/ttyACM0'.
Nov 06 15:03:50 wadew systemd[1]: esp32-auto-program@sys-devices-pci0000:00-0000:00:14.0-usb1-1\x2d2-1\x2d2:1.0-tty-ttyACM0.service: Main process exited, code=exited, status=1/FAILURE
Nov 06 15:03:50 wadew systemd[1]: esp32-auto-program@sys-devices-pci0000:00-0000:00:14.0-usb1-1\x2d2-1\x2d2:1.0-tty-ttyACM0.service: Failed with result 'exit-code'.
```

**Comment by tayken** [comment]
This may be a false alarm due to old things remaining from my tests. I was testing things out today and I got:

```bash
○ esp32-auto-program@sys-devices-pci0000:00-0000:00:14.0-usb1-1\x2d2-1\x2d2:1.0-tty-ttyACM0.service - Program ESP32 device attached to 'sys/devices/pci0000:00/0000:00:14.0/usb1/1-2/1-2:1.0/tty/ttyACM0'
     Loaded: loaded (/etc/systemd/system/esp32-auto-program@.service; static)
     Active: inactive (dead) since Mon 2024-11-18 21:29:42 JST; 29s ago
    Process: 3697 ExecStart=/home/tayken/ESP32-Auto-Programmer/build/esp32-program-shim.sh sys/devices/pci0000:00/0000:00:14.0/usb1/1-2/1-2:1.0/tty/ttyACM0 (code=exited, status=0/SUCCESS)
   Main PID: 3697 (code=exited, status=0/SUCCESS)
        CPU: 1.421s

Nov 18 21:28:48 wadew systemd[1]: Started Program ESP32 device attached to 'sys/devices/pci0000:00/0000:00:14.0/usb1/1-2/1-2:1.0/tty/ttyACM0'.
Nov 18 21:29:42 wadew systemd[1]: esp32-auto-program@sys-devices-pci0000:00-0000:00:14.0-usb1-1\x2d2-1\x2d2:1.0-tty-ttyACM0.service: Deactivated successfully.
Nov 18 21:29:42 wadew systemd[1]: esp32-auto-program@sys-devices-pci0000:00-0000:00:14.0-usb1-1\x2d2-1\x2d2:1.0-tty-ttyACM0.service: Consumed 1.421s CPU time.
```

So, seems like things are running OK.

**Comment by tayken** [comment]
Tried on my Raspberry Pi 3B and got failure again. I saw `DEVNAME` pop up once before. But again, programming is successful.

```
× esp32-auto-program@sys-devices-platform-soc-3f980000.usb-usb1-1\x2d1-1\x2d1.4-1\x2d1.4:1.0-ttyUSB0.service - Program ESP32 device attached to 'sys/devices/platform/soc/3f980000.usb/usb1/1-1/1-1.4/1-1.4:1.0/tt>
     Loaded: loaded (/etc/systemd/system/esp32-auto-program@.service; static)
     Active: failed (Result: exit-code) since Tue 2024-11-19 23:30:34 JST; 1min 9s ago
   Duration: 46ms
    Process: 12546 ExecStart=/home/pi/ESP32-Auto-Programmer/build/esp32-program-shim.sh sys/devices/platform/soc/3f980000.usb/usb1/1-1/1-1.4/1-1.4:1.0/ttyUSB0 (code=exited, status=1/FAILURE)
   Main PID: 12546 (code=exited, status=1/FAILURE)
        CPU: 46ms

Nov 19 23:30:34 raspberrypi-3b systemd[1]: Started esp32-auto-program@sys-devices-platform-soc-3f980000.usb-usb1-1\x2d1-1\x2d1.4-1\x2d1.4:1.0-ttyUSB0.service - Program ESP32 device attached to 'sys/devices/plat>
Nov 19 23:30:34 raspberrypi-3b esp32-program-shim.sh[12546]: /home/pi/ESP32-Auto-Programmer/build/esp32-program-shim.sh: line 5: DEVNAME: unbound variable
Nov 19 23:30:34 raspberrypi-3b systemd[1]: esp32-auto-program@sys-devices-platform-soc-3f980000.usb-usb1-1\x2d1-1\x2d1.4-1\x2d1.4:1.0-ttyUSB0.service: Main process exited, code=exited, status=1/FAILURE
Nov 19 23:30:34 raspberrypi-3b systemd[1]: esp32-auto-program@sys-devices-platform-soc-3f980000.usb-usb1-1\x2d1-1\x2d1.4-1\x2d1.4:1.0-ttyUSB0.service: Failed with result 'exit-code'.
```

**Comment by tayken** [comment]
The second issue was unrelated to the first one, we were starting the service for two devices. It's fixed with ef2f440.

There's no issue based on my checks, so I'll close this.
